app/src/main/java/com/jerboa/ui/components/blocks/BlockedPersonViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package com.jerboa.ui.components.blocks
+
+import android.content.Context
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.CreationExtras
+import com.jerboa.api.API
+import com.jerboa.api.ApiState
+import com.jerboa.api.toApiState
+import com.jerboa.feat.showBlockPersonToast
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockPerson
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockPersonResponse
+import it.vercruysse.lemmyapi.v0x19.datatypes.PersonId
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class BlockedPersonViewModel(
```

Comment:
Didn't have to make a separate Blocked Person/Community/Instance  viewmodel

Reply:
Agree, you can just use a single blockviewmodel, and have the functions pass in the id, and the block.